app/assets/javascripts/google_analytics.js.erb

```diff
@@ -0,0 +1,3 @@
+$(document).on('turbolinks:load', function () {
+  gtag('config', '<%= ENV['GOOGLE_ANALYTICS_ID'] %>', { 'page_path': window.location.pathname });
```

Comment:
I'm avoiding using .erb for assets. how about this?
```
if (typeof gtag === 'undefined') { return; }
  gtag('config', GOOGLE_ANALYTICS_ID, {
    'page_title' : document.title,
    'page_path': window.location.pathname
  });

```

Reply:
we also need to check if gtag is a function or not.

Reply:
Question: why would you avoid using `erb` for assets? The asset pipeline will always execute the extensions in order, so there's no technical reason not to use it.

Good catch re checking for being a function.